starlette/datastructures.py

```diff
@@ -1,19 +1,35 @@
+import sys
 import tempfile
 import typing
-from collections import namedtuple
-from collections.abc import Sequence
 from shlex import shlex
 from urllib.parse import SplitResult, parse_qsl, urlencode, urlsplit
 
+if sys.version_info >= (3, 8):  # pragma: no cover
+    from typing import Protocol
+else:  # pragma: no cover
+    from typing_extensions import Protocol
+
 from starlette.concurrency import run_in_threadpool
 from starlette.types import Scope
 
-Address = namedtuple("Address", ["host", "port"])
+_T = typing.TypeVar("_T")
+_KT = typing.TypeVar("_KT")  # key type
+_VT = typing.TypeVar("_VT")  # value type
+_KT_co = typing.TypeVar("_KT_co", covariant=True)  # key type for covariant containers
+_VT_co = typing.TypeVar("_VT_co", covariant=True)  # value type for covariant containers
+
+
+class Address(typing.NamedTuple):
+    host: typing.Optional[str]
+    port: typing.Optional[int]
 
 
 class URL:
     def __init__(
-        self, url: str = "", scope: Scope = None, **components: typing.Any
+        self,
+        url: str = "",
+        scope: typing.Optional[Scope] = None,
```

Comment:
```suggestion
        scope: Scope = None,
```

Reply:
Could you explain this change request to me please, or reference some documentation on the subject? I understand that there is both a practical and semantic difference, but as far as I understand using `Scope | None` is now preferred if the default value is `None`. For example, see https://stackoverflow.com/questions/62732402/can-i-omit-optional-if-i-set-default-to-none